### ISSUE TITLE
Range Slider JS not working for dynamic content

### ISSUE THREAD

**Issue by punter82** [andreruffert/rangeslider.js#208]
Hey Everyone,

 I am using the basic range slider and i have written like how it is mentioned in the docs. It works fine on a page which is rendered. Basically when i view source i see the content.

But for scenarios like modal, or angular ng-view where the content is dynamically added, the range slider is not initialized and i see the default html5 slider. No errors nothing. Infact i tried initiating after opening the modal and i see the object but the slider is not changed.

Basically all the html elements needed for range slider is not appended. Any help is highly appreciated

**Comment by punter82** [comment]
I would like to know if this is a known issue or if someone has faced it and has a solution ?

**Comment by punter82** [comment]
Can someone provide some help. It is kinda urgent..

**Comment by andreruffert** [comment]
K first of all keep calm man :v: 

Of course u have to trigger a initialization for dynamically added content.
Rangeslider.js doesn't watch for DOM changes.

So you need to watch for DOM changes and trigger the rangeslider.js initialization.
It doesn't have really something to do with rangeslider.js itself...

```js
$('input[type="range"]', 'addedContentParentSelector').rangeslider({...});
```

**Comment by punter82** [comment]
i did try it but didnt work.. Anyways will try something different.

**Comment by andreruffert** [comment]
K I think you did something wrong.
It's kinda hard to tell you a possible approach without seeing any specific code or knowing exactly what you'r trying to achieve... 

Good luck anyway! :v:

**Comment by punter82** [comment]
I made it work.. All good

**Comment by agseco** [comment]
@punter82 I'm having a similar issue to yours, any chance that you share how you solved your issue?

**Comment by punter82** [comment]
Sure. Ours is an angular JS app. So my scenario was, we have ngDialog modal where people can slide and rate, but the ngDialog was not there initially in the DOM. So i wrote a callback to detect if ngDialog was successfully opened and then the function does something like this
 var $r = jQuery('input[type="range"]');
            // Initialize
            $r.rangeslider({
                polyfill: false,
                onInit: function() {
                   // Do some logic on the plugin init
                },
                onSlideEnd: function(position, value) {
                    // Update the rating value once the slide is done

                }
            });

Worked like Charm.

**Comment by abnayak** [comment]
Any idea how I can reinitialize rangeslider for below code which is adding range elements dynamically using Vue.js?

``` html
<ol>
 <li v-for="light in lights">
    <input type="range" min="0" max="255" step="1">
</li>
</ol>
```
